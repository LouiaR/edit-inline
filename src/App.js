import React from "react";
import { GlobalStyle } from "./components/style";
import styled from "styled-components";

import CoyoteLogo from "./assets/CSoftware_logo_Neg.svg";

const Logo = styled.img`
  padding: 1em 0;
  display: inline-block;
  text-decoration: none;
  img {
    width: 11em;
  }

  @media (max-width: 48em) {
    padding: 1em;
  }
`;

function App() {
  return (
    <div className="App">
       <GlobalStyle />
       <Logo src={CoyoteLogo} alt="logo" />
    </div>
  );
}

export default App;
