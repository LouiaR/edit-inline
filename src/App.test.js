import React from "react";
import { shallow  } from "enzyme";
import toJson from "enzyme-to-json";


import CoyoteLogo from "./assets/CSoftware_logo_Neg.svg";
import App, { Logo } from "./App";

describe("rendering components", () => {
  it("renders App component without crashing", () => {
    shallow(<App />);
  });
  it("renders App component logo without crashing", () => {
    const wrapper = shallow(<App />);
    const header = <Logo src={CoyoteLogo} alt="logo" />;
    expect(wrapper.contains(header)).toEqual(true);
  });
});

describe("snapshots", () => {
  it("App snapshot", () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
