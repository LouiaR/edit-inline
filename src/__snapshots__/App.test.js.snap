// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots App snapshot 1`] = `
<AppContextProvider
  value={
    Object {
      "setState": [Function],
      "state": Object {
        "error": false,
        "isEditing": false,
        "isLoading": false,
        "message": "",
        "text": "Hello World",
      },
    }
  }
>
  <Memo(l) />
  <styled.img
    alt="logo"
    src="CSoftware_logo_Neg.svg"
  />
  <Editable />
</AppContextProvider>
`;
